docs/building.md

```diff
@@ -18,90 +18,85 @@ a number of hours to complete.
-    $ git submodule update --recursive
+    $ git submodule update --init --recursive
   ```
 2. Install Rust: install [via rustup](https://www.rust-lang.org/tools/install)
-3. Install your system dependencies: - install via the instructions below for [Linux](building.md#linux), [MacOS](building.md#macos) or [Windows](building.md#windows)
-4. Check dependencies, environment variables and test
-   1. Run: `./libs/verify-desktop-environment.sh`
-   1. Run: `cargo test`
-
-Once you have successfully run `./libs/verify-desktop-environment.sh` and `cargo test` you can move to the **Building for Fenix** and **Building for iOS** sections below to setup your local environment for testing with our client applications.
-
-
-#### Linux
-1. Install the system dependencies required for building NSS
-  1. Install gyp: `apt install gyp` (required for NSS)
-  1. Install ninja-build: [via package for distribution](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages#package-managers)
-  1. Install python3: [3.6 via python.org](https://docs.python.org/3/using/unix.html)
-  1. Install zlib: `apt install zlib1g-dev`
-1. Install the system dependencies required for SQLcipher
-  1. Install tcl: `apt install tclsh` (required for SQLcipher)
-
-
-#### MacOS
-1. Install Xcode: check the [ci config](../.circleci/config.yml) for the correct
-version.
-1. Install Xcode tools: `xcode-select --install`
-1. Install homebrew: [via homebrew](https://brew.sh/) (its what we use for ci)
-1. Install the system dependencies required for building NSS
-    1. Install ninja: `brew install ninja`
-    1. Install gyp (via https://github.com/mogemimi/pomdog/wiki/How-to-Install-GYP)
-1. Install swift-protobuf: `brew install swift-protobuf`
-
-
-#### Windows
-*Install windows build tools*
-
-> Why [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/about)?
->
-> It's currently tricky to get some of these builds working on Windows, primarily due to our use of SQLcipher. By using WSL it is possible to get builds working, but still have them published to your "native" local maven cache so it's available for use by a "native" Android Studio.
-
-1. Install [WSL](https://docs.microsoft.com/en-us/windows/wsl/about) (recommended over native tooling)
-1. Install unzip: `sudo apt install unzip`
-1. Install python3: `sudo apt install python3` *Note: must be python 3.6*
-1. Install system build tools: `sudo apt install build-essential`
-1. Install zlib: `sudo apt-get install zlib1g-dev`
-1. Install tcl: `sudo apt install tcl-dev`
+3. Install your system dependencies:
+
+    #### Linux
+    1. Install the system dependencies required for building NSS
+        1. Install gyp: `apt install gyp` (required for NSS)
+        1. Install ninja-build: [via package for distribution](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages#package-managers)
+        1. Install python3: [3.6 via python.org](https://docs.python.org/3/using/unix.html)
```

Comment:
small nit, but I don't think we rely on exactly 3.6 - eg, I have 3.8. I wonder if we can soften this to say something like "at least 3.6"? (That's tricky because I suspect 3.10 will not work for obscure reasons, so feel free to do with this comment as you like)